packages/mui-material-next/src/Option/Option.tsx

```diff
@@ -0,0 +1,321 @@
+'use client';
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { OverridableComponent } from '@mui/types';
+import { alpha } from '@mui/system';
+import {
+  unstable_useEnhancedEffect as useEnhancedEffect,
+  unstable_useForkRef as useForkRef,
+} from '@mui/utils';
+import { useSlotProps } from '@mui/base/utils';
+import { unstable_composeClasses as composeClasses } from '@mui/base/composeClasses';
+import { useMenuItem } from '@mui/base/useMenuItem';
```

Comment:
Out of curiosity, is semantically correct to use the useOption here? I suppose if yes, it will come in a follow up PR. Just making sure :)

Reply:
Yes! `Option` will use `useOption`, [it's coming in the follow up PR](https://github.com/mui/material-ui/pull/40210/files#diff-2bc38a6fd18829455bbb4fc7fd99e2985dfa840e93835e87294d5e7bfd213fedR12).